### DISCUSSION TITLE
How to deprecate a type

### DISCUSSION THREAD

**Discussion by mrFatDZ** [kamilkisiela/graphql-inspector#1827] [General]
Hello,

How do we correctly deprecate a type without raising an error by the inspector? 

GraphQL deprecated directive is only allowed on fields and enums but we want to remove an entire type since we no longer use it in our schema.

Is there a correct path we can take such that the rule "suppressRemovalOfDeprecatedField" marks type removal as a warning instead of an error? 

Thanks!

**Comment by kamilkisiela** [comment]
You can use a custom directive + custom validation rule https://github.com/kamilkisiela/graphql-inspector/blob/ed0503a711bc9e99276845b2f7b573ee14970f15/packages/commands/diff/src/index.ts#L217
